Add blockhound exceptions for log4j internal logger registry

### Describe the bug

```
java.lang.AssertionError: Blocking call! jdk.internal.misc.Unsafe#park on thread Thread[#275,
non-blocking-thread-EmbeddedRestSearchTest-NodeA-3-6,5,ISPN-non-blocking-group]
2025-09-11T04:58:26.8618370Z    at org.infinispan.util.CoreTestBlockHoundIntegration.lambda$applyTo$0(CoreTestBlockHoundIn
tegration.java:51)
2025-09-11T04:58:26.8620348Z    at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488)
2025-09-11T04:58:26.8621705Z    at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89)
2025-09-11T04:58:26.8622871Z    at java.base/jdk.internal.misc.Unsafe.park(Unsafe.java)
2025-09-11T04:58:26.8623989Z    at java.base/java.util.concurrent.locks.LockSupport.park(LockSupport.java:221)
2025-09-11T04:58:26.8625464Z    at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:754)
2025-09-11T04:58:26.8627197Z    at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:990)
2025-09-11T04:58:26.8629051Z    at java.base/java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.lock(ReentrantReadWriteLock.java:959)
2025-09-11T04:58:26.8630842Z    at org.apache.logging.log4j.core.util.internal.InternalLoggerRegistry.computeIfAbsent(InternalLoggerRegistry.java:235)
2025-09-11T04:58:26.8639892Z    at org.apache.logging.log4j.core.LoggerContext.getLogger(LoggerContext.java:588)
2025-09-11T04:58:26.8646535Z    at org.apache.logging.log4j.core.LoggerContext.getLogger(LoggerContext.java:561)
2025-09-11T04:58:26.8648021Z    at org.apache.logging.log4j.core.LoggerContext.getLogger(LoggerContext.java:71)
2025-09-11T04:58:26.8649744Z    at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:624)
2025-09-11T04:58:26.8651588Z    at io.netty.util.internal.logging.Log4J2LoggerFactory.newInstance(Log4J2LoggerFactory.java
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/core/src/test/java/org/infinispan/util/CoreTestBlockHoundIntegration.java
+++ b/core/src/test/java/org/infinispan/util/CoreTestBlockHoundIntegration.java
@@ -3,6 +3,7 @@ package org.infinispan.util;
 import java.io.File;
 import java.io.IOException;
 
+import org.apache.logging.log4j.core.util.internal.InternalLoggerRegistry;
 import org.infinispan.commons.IllegalLifecycleStateException;
 import org.infinispan.commons.internal.CommonsBlockHoundIntegration;
 import org.infinispan.commons.test.PolarionJUnitTest;
@@ -116,6 +117,8 @@ public class CoreTestBlockHoundIntegration implements BlockHoundIntegration {
       // The join is used to allow for a sync API for test simplicity - where as the actual store invocation
       // must be non blocking
       builder.allowBlockingCallsInside(WaitNonBlockingStore.class.getName(), "join");
+      // Log4j's logger factory
+      builder.allowBlockingCallsInside(InternalLoggerRegistry.class.getName(), "computeIfAbsent");
    }
 
    private static void writeJUnitReport(String testName, Throwable throwable, String type) {

